### ISSUE TITLE
On Seed Creation you get an infinite loading screen.

### ISSUE THREAD

**Issue by Suzamoon** [Doug-Murphy/LethalCompanyMods#37]
Seems that when assigning values on V50 to everything and a seed is selected for a mood I'm unable to make it past the loading screen/seed selection.

**Comment by Suzamoon** [comment]
This is specifically for your mod Balanced Items.

**Comment by Doug-Murphy** [comment]
I'm sorry but I'm really confused about what the issue is. Here's what I assume you mean:

When playing v50 and assigning a value to (some particular feature of BI) in the config file, and pulling the lever to load a map (when the seed is shown) your game freezes.

Is that correct?

I just tried to reproduce this issue and was unable to. Everything seems to be working fine for me. I set custom values for weights of items, all features are enabled, I set rarity overrides for Adamance and I landed on Adamance. I had no issues on my end.

If you have an error message or something in your log file, I can take a quick look at that and hopefully help you from there. For now, I'm closing this issue as "can't repro".

**Comment by Suzamoon** [comment]
I was able to reproduce it upon setting the config features to 
![Screenshot 2024-04-28 222110](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/a1fa9c14-deda-4ae0-90bf-3255752e5c3f)
Seemingly immediately it was throwing this error
![Screenshot 2024-04-28 222256](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/852784f7-e66f-45f4-8228-80092b901790)
Which I suspect is related to the NeedyCats mod by Jordo.
Could it be due to not being on the list of vanilla items it fails to define and then hangs on generating a seed?

**Comment by Doug-Murphy** [comment]
Try the steps below to see if this fixes it. I have a theory. 

1. Close the game. 
2. Set the moon rarity override to true. 
3. Launch the game and start a round (any moon should be fine)
4. Confirm there is no issue. 
5. Exit to main menu - DO NOT CLOSE THE GAME
6. Set the moon rarity override to false. 
7. Reload the save game, or start a new one. 
8. Start a round (any moon should be fine)
9. Confirm there is no issue.

**Comment by Suzamoon** [comment]
Replicated the steps above
1. Closed the game
2. Set Moon rarity Override to true
3. Launched game and started a round
4. Successful round start
5. Exit to main menu
6. Edit Config to Moon Rarity Override to false
7. Reload save game
8. Failed to start a round.

**Comment by Doug-Murphy** [comment]
Interesting. I followed the steps above and did not have any issue. It is possible that you have a broken config file, or some sort of mod incompatibility. CatItem is, after all, not a vanilla item. :D I will assume it's that mod where you find cats and they purr and you can pet them.

Can you get another screenshot of your log output? This time make sure you show the surrounding log entries as well instead of just the red error message. Ensure that you have set the LogLevels of BepInEx.cfg to "All" for the "Logging.Console" section.
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/1383347/ce92d234-39fc-4f7d-a894-c36dc33e821e)

Anywho, it seems likely that this is probably a bad config file. The error you're seeing is when the mod is trying to fetch the rarity override for an item on the moon that you're landing on. It only attempts to fetch for items that are found to be spawnable on that moon, and all items _should_ exist in that dictionary that it's setting from because that dictionary is built when the round starts. It loops through all moons and all items in your game (vanilla + modded) and initializes them all at that point. :) Then when you pull the lever to land the ship, it updates the game's values to what you want them to be.

Open up your .cfg file, find "CatItem" and paste those entries for me here. There should be several entries for it - once for sell value, once for weight, and once _per moon it can spawn on_ for the rarity override.

For example:
```
## The average sell value for the 'Bell' item. The default vanilla value is '160'.
# Setting type: UInt16
# Default value: 160
# Acceptable value range: From 0 to 65535
Bell = 160


## The rarity for the 'Bell' item. The default vanilla value is '37'.
# Setting type: Int32
# Default value: 37
# Acceptable value range: From 0 to 100
Bell = 37


## The weight for the 'Bell' item. The default vanilla value is '23'.
# Setting type: Single
# Default value: 5
# Acceptable value range: From 0 to 1000
Bell = 5
```

**Comment by Suzamoon** [comment]
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/e80e06ab-c20f-462b-b051-44ceb9fc8c55)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/c9c62fd8-2059-4b98-a880-32b3e2f9d52b)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/11b70f35-3fbb-495c-ae24-08ad9a0e9311)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/c424651a-b276-45d4-81cc-50184b8b656c)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/51cf2fce-eec8-4437-900b-5fb4fbeb309d)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/94064cd7-0df6-4eff-91e3-e4dd80c6104d)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/6fe8ca3c-ab20-4aed-b43a-dd76a3b739dc)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/58c6e575-e0b2-433b-beed-4b9b1f0d4f36)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/1f987c3b-69dd-4611-a17c-06e2411aa826)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/2131b384-5a48-4277-8f9f-d83f1e7a4337)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/43806a84-9dde-4ac8-b6e3-b8511a08450e)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/5643f83f-b2af-4e89-9e0c-7c17f7230a37)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/8c57f0af-bc42-4d28-8169-64e286d0bad3)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/df50ee3e-697a-4840-9de8-9b66cf214dff)
Here are the Values from the config

**Comment by Suzamoon** [comment]
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/97ab9468-7cf8-4999-a7a3-096d2e742c45)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/389373cf-157a-49bd-b2f6-d8758d3619b7)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/80849e68-b1cc-42a0-9b87-0b54ebf2257d)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/a030b48f-b78b-467e-87b3-8cbcd9a71d15)
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/51769303/d051a56b-6303-48d5-b554-d1728f678f28)
Here is the log upon attempting to generate a level.

**Comment by Doug-Murphy** [comment]
I see. These are the interesting pieces of information that you provided.

This screenshot is of valid config entries. Note that it has no comments describing it.
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/1383347/d65f3dfd-b8fb-4bd0-9cb7-09e400c2fcaa)

This screenshot is of INVALID config entries. Not that it has no comments describing it. I bet that for other items on this moon and other moons, there are comments for it.
![image](https://github.com/Doug-Murphy/LethalCompanyMods/assets/1383347/37a9b57f-8647-43d4-a5b0-3f1571e5c858)

This also explains why the issue is only regarding moon rarities for you. The comments are missing only for the moon rarity sections. Why are these entries invalid? Well that is not evident here (and would not be worth chasing down). But your config file is kinda sorta broken but not entirely broken. The times that I have seen my config file get into this state with testing was _I THINK_ when I was toying around with adding and removing mods from my modpack. Either way, it's definitely an oddity with how BepInEx parses and binds the config files. Proprietary format is a bit wonky. :) It is a bit easier to read and modify manually, but I'd say the vast majority are using a tool like r2 that reads the .cfg file entries for you. And that tool could just as easily do that with XML or JSON which would be much less error prone.

For the solution, you have 2 options:
1. Delete your .cfg file and start with a fresh one.
2. Delete the offending entries from your .cfg file. (This _should_ work, but I don't guarantee it.)

Whichever one you want to try is really up to you, but that's really the best I can offer for solutions here.

In summary, BepInEx does funky things with config files due to proprietary format. Adding and removing mods can kinda trip it up because I do not DELETE entries from your .cfg file, ever. That's a conscious decision that I made so that a temporary disabling/removing of a mod would not result in you also losing your overrides that you set for it.

**Comment by Suzamoon** [comment]
So upon trying to

1. Delete your .cfg file and start with a fresh one.

-  It interestingly still did not solve the issue. However, I did notice that when I went to investigate the .cfg file it did not generate entries for rarities this time. Only weight and value.

2. Delete the offending entries from your .cfg file.

- I also attempted to do this by manually going into the config file and editing it outside of R2/Thunderstore etc and the same error still persisted. It actually still threw the error referencing the cat item rarity log from earlier.

3. Manually adding the items to the .cfg file?

- Still actually wound up with the same error referencing the cat item rarity. I'm actually curious as to what would happen if I found a cat with it running now and attempted to sell it etc.

Mainly was just adding this as more food for thought.
